### ISSUE TITLE
seq2seq output not as expected

### ISSUE THREAD

**Issue by student9292** [keras-team/keras#10478]
I posted my question in keras.io and stackoverflow, but could not reach someone for help. I am dealing with Seq2Seq prediction and have the following problem: 

I implmented the keras seq2seq model on timeseries data. I will not post a code because I think it is not a coding problem but the code can be found here (scroll down for the post) or look here The model still gives good results on data which are far away from the training data. Why is this happening ?
I used already different hidden units sizes, epochs, batchsize and timesteps parameters, but I train with a range of -0.2 to 0.2 and did not expect that the model can predict also values till -0.8 to 0.8 back, which is happening. So it not only generalize from the data, its more an adjustable prediction, which should not happen.

    num_encoder_tokens = 1
    num_decoder_tokens = 1
    encoder_seq_length = None
    decoder_seq_length = None
    batch_size = #100
    epochs = #1
    hidden_units= #30
    timesteps= #20

    #Input Data 
     input_seqs=train #data with sliding window, shape(None,20,1)
     target_seqs=reversed_train #first reversed than sliding window added 
     with same shape as above

     #define training encoder
     encoder_inputs = Input(shape=(None, num_encoder_tokens))
     encoder = LSTM(hidden_units, return_state=True)
     encoder_outputs, state_h, state_c = encoder(encoder_inputs)
     encoder_states = [state_h, state_c]
     #define training decoder
     decoder_inputs = Input(shape=(None,num_decoder_tokens))
     decoder_lstm = LSTM(hidden_units, return_sequences=True, 
     return_state=True)

     decoder_outputs, _, _ = decoder_lstm(decoder_inputs, 
     initial_state=encoder_states)
     decoder_dense = Dense(num_encoder_tokens, activation='tanh')
     decoder_outputs = decoder_dense(decoder_outputs)
     model = Model([encoder_inputs, decoder_inputs], decoder_outputs)

     #training
     model.compile(optimizer='adam', loss='mse')
     model.fit([input_seqs, target_seqs], target_seqs,batch_size=batch_size, 
      epochs=epochs,validation_split=0.2)

     #Testdata
     target_seqs=test # different data as target with same shape

     #define inference encoder
     encoder_model = Model(encoder_inputs, encoder_states)
     #define inference decoder
     decoder_state_input_h = Input(shape=(hidden_units,))
     decoder_state_input_c = Input(shape=(hidden_units,))
     decoder_states_inputs = [decoder_state_input_h, decoder_state_input_c]

     decoder_outputs, state_h, state_c = decoder_lstm(decoder_inputs, 
     initial_state=decoder_states_inputs)
     decoder_states = [state_h, state_c]
     decoder_outputs = decoder_dense(decoder_outputs)
     decoder_model = Model([decoder_inputs] + decoder_states_inputs, 
     [decoder_outputs] + decoder_states)

     #Initalize states for decoder 
     states_values = encoder_model.predict(input_seqs)

     #empty target
     target_seq = np.zeros((1, 1, num_decoder_tokens))
     #predict
     output=list()
     for t in range(timesteps):
        output_tokens, h, c  = decoder_model.predict([target_seqs] + states_values)
        output.append(output_tokens[0,0,:])
        states_values = [h,c]
        target_seq = output_tokens

#with this easier model with or without repeatvector does not mattter, I also get the same problem: 

    from keras.models import Sequential, Model
    from keras.layers import LSTM, RepeatVector

    model = Sequential()
    model.add(LSTM(40, input_shape=(timesteps, n_features)))
    model.add(RepeatVector(timesteps))
    model.add(LSTM(40, return_sequences=True))
    model.add(Dense(1, activation='tanh'))
    model.compile(loss='mse', optimizer='adam', metrics=['mae'])
    history=model.fit(train, train, epochs=200, shuffle=False)
